### ISSUE TITLE
allow to show load in ms instead of %

### ISSUE THREAD

**Issue by axet** [flightlessmango/MangoHud#788]
hello!

let us to show cpu and gpu load in miliseconds instead of % load per frame. for example: cpu 2ms; gpu 4ms; frame 4ms; 

right now it showing useless stats like: cpu 1% gpu 4%; fps 60 16ms (i know 60fps is 16ms)

EDIT:

You can easily get those values. For example CPU 5% GPU 18% FPS 60 == 16ms

frame rate = 60fps (16ms)
cpu load 5% = 16ms * 5% = CPU 0.8 ms
gpu load 20% = 16ms * 20% = GPU 3.2 ms
frame time = max(cpu ms, gpu ms) = 2.88 ms

Lest's assume we have 100% cpu load.

frame rate = 16fps (62.5ms)
cpu load = 100% = 62.5ms
gpu load = 20% = 12.5ms
frame time = 62.5ms

Best option would be to know frame time / idle time somehow else.

**Comment by flightlessmango** [comment]
Is that how gpu/cpu ms works? it's just relative to the frame rate?

**Comment by axet** [comment]
No. Proper way is more complicated and requires measuring the actual function calling time. But here you have no such option available and can simplify to the CPU / GPU load.

**Comment by flightlessmango** [comment]
Through /proc we do have cpu usage for the app, we even have gpu usage for the app for amd and intel.
I assume we could figure this out through the nvml library as well.
So it does seem possible to figure out the actual values

**Comment by mikkorantalainen** [comment]
If you look at generic measurements, you'll never have truly accurate stats so representing these as milliseconds might be misleading. In reality, GPU time and CPU time are interleaved, something like this (ascii graphics):
```
| < start of rendering             end of rendering >|
========== CPU ACTIVE ==============
                ========= GPU ACTIVE =================
```

Typically the CPU must do something related to physics simulation first for the whole world, then it starts to stream commands to GPU which executes them in parallel while CPU is computing more stuff to render.

However, generic interfaces cannot tell the idle period of the GPU. It might like above but it might be like this, too:

```
| < start of rendering             end of rendering >|
========== CPU ACTIVE ========  ==   == = =
== == ==   === GPU ACTIVE ==  =   ===   == ==  =======
```

That is, identical total CPU and GPU active use within a single frame but lots of little idle moments for both. The true bottleneck could be somewhere else in the system but rendering this as "cpu ms" and "gpu ms" could give incorrect belief that gpu and cpu were 100% loaded for this time. But the true bottleneck could be some kind of lock contention in the system and it's already running as fast as possible because lock contention turns parallel processing into sequential processing in reality. And depending on the generic interface you use, lock contention might even count as busy time instead of idling.

If the available API only reports usage percentage, it would be better to render the percentage because MangoHud cannot know any better in reality.
